app/controllers/change_answers_controller.rb

```diff
@@ -9,34 +10,21 @@ def update
       # Note that this merge! has the side effect of altering the session data in the '@check' variable
       # that was loaded by our base class which looks quite awkward
       session_data.merge!(@form.attributes_for_export_to_session)
-      if @check.consistent?
-        if Steps::Helper.last_step_in_group?(session_data, step)
-          # if we have a 'check stops' block, it may have been removed even if we're consistent
-          # e.g. going from employed to unemployed
-          # but only check this if we have financial information in the check
-          if Steps::Logic.check_stops_at_gross_income?(session_data) && !Steps::Logic.non_means_tested?(session_data)
-            last_step_with_data = Steps::Helper.last_step_with_valid_data(session_data)
-            completed_steps = Steps::Helper.completed_steps_for(session_data, last_step_with_data)
-            cfe_result = CfeService.result(session_data, completed_steps)
-
-            if cfe_result.ineligible_gross_income?
-              # no change - go back to check answers
-              save_and_redirect_to_check_answers
-            else
-              # eligibility changed - remove 'ineligible' block and show notification
-              session_data.delete IneligibleGrossIncomeForm::SELECTION
-              flash[:notice] = I18n.t("service.change_eligibility")
-              redirect_to_next_question
-            end
-          else
-            save_and_redirect_to_check_answers
-          end
+      if FeatureFlags.enabled?(:ee_banner, session_data)
+        next_step = step_with_inconsistent_data
+        if next_step
+          redirect_to helpers.check_step_path_from_step(next_step, assessment_code)
+        elsif Steps::Helper.last_step_in_group?(session_data, step)
```

Comment:
might not need this branch anymore

Reply:
thus removes so much complexity 👏

Reply:
confirmed that this branch is still needed for now